### DISCUSSION TITLE
Dealing with handlers requiring a HandlerFunc; within ogen?

### DISCUSSION THREAD

**Discussion by johannes-riecken** [ogen-go/ogen#1002] [Q&A] [answered]
For health status handlers we use a company-internal library which provides a HandlerFunc.
The easiest way to make use of it is to use mux.Handle, as described in the ogen docs, but as we have the path also defined in the OpenAPI spec, ogen generates code which requires to implement the handler matching its Handler interface, and I don't see how to convert our HandlerFunc to that shape. In our OpenAPI spec, the health check follows a JSON schema, where for keys like "database" values like "OK", "WARN" and "ERROR" are possible.

Do you see a way to get around this problem? Would it make sense for ogen to add support for another annotation, something like `x-ogen-ignore`, to skip code generation for one path?

**Comment by ghost** [comment, marked as answer]
You can use `--filter-path` option to exclude some `PathItem`s from generated code.

**Reply by wolfsblu** [reply]
Unfortunately this option does not seem to exist anymore, and the approach of using the YAML config with `path_regex` fails when trying to exclude specific paths, as it does not support negative lookahead.
